File: KadirliApp.Domain/Entities/GuideItem.cs
using KadirliApp.Domain.Common;

namespace KadirliApp.Domain.Entities;

public class GuideItem : BaseEntity
{
    public Guid CategoryId { get; set; }
    public string Name { get; set; } = default!;
    public string? Phone { get; set; }
    public string? Address { get; set; }
    public string? Email { get; set; }
    public string? WebsiteUrl { get; set; }
    public string? WorkingHours { get; set; }
    public decimal? Latitude { get; set; }
    public decimal? Longitude { get; set; }
    public Guid? LogoFileId { get; set; }
    public string? Description { get; set; }
    public bool IsActive { get; set; } = true;

    public GuideCategory Category { get; set; } = default!;
    public File? LogoFile { get; set; }
}

File: mobile/lib/features/guide/data/models/guide_item.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'guide_item.freezed.dart';
part 'guide_item.g.dart';

/// `GET /v1/guide/items` ve `/v1/guide/items/{id}` gövdesi (`GuideItemDto`).
///
/// Kategori bilgisi satırın içinde geliyor (`categoryName`) — liste ekranı
/// ayrıca kategori çekmek zorunda değil.
@freezed
abstract class GuideItem with _$GuideItem {
  const factory GuideItem({
    required String id,
    required String name,
    String? categoryId,
    String? categoryName,
    String? categoryIcon,
    String? categoryColor,
    String? phone,
    String? address,
    String? email,
    String? websiteUrl,
    String? workingHours,
    double? latitude,
    double? longitude,
    @Default(false) bool hasLocation,
    String? description,
    @Default(true) bool isActive,
  }) = _GuideItem;

  const GuideItem._();

  factory GuideItem.fromJson(Map<String, dynamic> json) =>
      _$GuideItemFromJson(json);

  bool get hasPhone => (phone ?? '').trim().isNotEmpty;

  /// Haritada gösterilebilir mi — sunucunun `hasLocation` bayrağı ile
  /// koordinatların **ikisi de** doğru olmalı (bayrak true ama koordinat null
  /// gelen kayıtlara karşı savunma).
  bool get canOpenMap => latitude != null && longitude != null;
}
